kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -280,7 +275,23 @@ protected void handleInactive(KafkaHeaderAndRequest kafkaHeaderAndRequest,
 
     // Leverage pulsar admin to get partitioned topic metadata
     private CompletableFuture<PartitionedTopicMetadata> getPartitionedTopicMetadataAsync(String topicName) {
-        return admin.topics().getPartitionedTopicMetadataAsync(topicName);
+        return admin.topics().getPartitionedTopicMetadataAsync(topicName).thenCompose(partitionedTopicMetadata -> {
```

Comment:
same as comments in line 331,  Would you please open a new PR to track the change of this to `pulsarService.getBrokerService().fetchPartitionedTopicMetadataAsync`?

Reply:
The change is mainly for transfer some work in `handleTopicMetadataRequest` to another method because `handleTopicMetadataRequest` is too long now (after the change it still has 200 lines).

Reply:
Yes, This comments may not related with this main change, we could open issue to do the change in the later PRs

Reply:
Ok, I would change it back and open a new PR to process these things.